src/bin/nydusd/main.rs

```diff
@@ -363,6 +373,83 @@ fn handle_rlimit_nofile_option(args: &ArgMatches, option_name: &str) -> Result<(
     Ok(())
 }
 
+fn send_unix_socket_request(socket_path: &str, method: &str, url: &str) -> Option<String> {
+    let mut stream = match UnixStream::connect(socket_path).ok() {
+        Some(stream) => stream,
+        None => return None,
+    };
+
+    let request = format!("{} {} HTTP/1.1\r\nHost: localhost\r\n\r\n", method, url);
```

Comment:
How about using the `hyper::Client` crate directly, instead of implementing the HTTP over the unix domain socket client by self?